extensions/control-plane/api/data-management/catalog-api/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/catalog/CatalogApiController.java

```diff
@@ -61,6 +64,26 @@ public void getCatalog(@jakarta.validation.constraints.NotNull(message = PROVIDE
             monitor.debug("No paging parameters were supplied, using 0...Integer.MAX_VALUE");
         }
 
+        performQuery(providerUrl, spec, response);
+    }
+
+    @Override
+    @POST
+    @Path("/request")
+    public void requestCatalog(@RequestBody(required = true) CatalogRequestDto requestDto, @Suspended AsyncResponse response) {
+
+        var query = QuerySpec.Builder.newInstance()
```

Comment:
Is it expected to not use a transformer for mapping the CatalogRequestDto to a QuerySpec?

Reply:
Same opinion, I'd say it's better to follow the pattern and introduce a transformer.

Reply:
yes, the reason I didn't yet do it, is because in my view the `CatalogRequestDto` is only here temporarily, as long as we have the `QuerySpecDto` and the `CatalogRequestDto` in parallel. They are very similar, the only difference is that the `QuerySpecDto` has a `String filterExpression`, whereas the `CatalogRequestDto` has a `List<Criterion> filter`. 

Once we have established the pattern to use a `POST` in all the endpoints, we can then modify the `QuerySpecDto` to have the `List<Criterion>`, by which point it would replace the `CatalogRequestDto`.

TL;DR: I was too lazy adding the transformer, because `CatalogRequestDto` is only a temporary class.

Reply:
@ndr-brt @bscholtes1A since the `CatalogRequestDto` now contains a `CriterionDto`, for which we already have a transformer, I added a `CatalogRequestDtoToQuerySpecTransformer` as well.